src/test/java/mate/academy/spring/boot/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,309 @@
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/add-cart-items-to-cart-items-table.sql"));
+        }
+    }
+
+    @BeforeEach
+    void setup() {
+        User mockUser = new User();
+        mockUser.setId(3L);
+        mockUser.setEmail("user1@example.com");
+
+        SecurityContextHolder.getContext().setAuthentication(
+                new UsernamePasswordAuthenticationToken(mockUser, null, List.of())
+        );
+    }
+
+    @AfterEach
+    void afterAll(
+            @Autowired DataSource dataSource
+    ) {
+        teardown(dataSource);
+    }
+
+    @AfterEach
+    void tearDown() {
+        SecurityContextHolder.clearContext();
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/remove-cart-items-from-cart-items-table.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/remove-shopping-carts-from-shopping-carts-table.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/remove-books-from-books-table.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/remove-users-from-users-table.sql"));
+        }
+    }
+
+    @Test
+    @DisplayName("Request to add book to ShoppingCart with valid request , we are waiting ShoppingCartDto")
+    public void testAddBookToCart_whenValidInput_shouldReturnShoppingCartDto() throws Exception {
+        Long userId = 3L;
+        CartItemRequestDto requestDto = new CartItemRequestDto();
+        requestDto.setBookId(2L);
+        requestDto.setQuantity(1);
+
+        ShoppingCartDto expected = new ShoppingCartDto();
+        expected.setId(3L);
+        expected.setUserId(userId);
+        CartItemDto cartItemDtoFirst = getCartItemDto(1L, 1L, "Book 1", 2);
+        CartItemDto cartItemDtoSecond = getCartItemDto(3L, 2L, "Book 2", 1);
+        Set<CartItemDto> cartItemDtoSet = new HashSet<>();
+        cartItemDtoSet.add(cartItemDtoFirst);
+        cartItemDtoSet.add(cartItemDtoSecond);
+        expected.setCartItems(cartItemDtoSet);
+
+        MvcResult result = mockMvc.perform(post("/cart")
+                        .contentType("application/json")
+                        .content(objectMapper.writeValueAsString(requestDto))
+                        .principal(() -> String.valueOf(userId)))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        ShoppingCartDto actual = objectMapper.readValue(result.getResponse().getContentAsString(), ShoppingCartDto.class);
+
+        assertNotNull(actual);
+        assertTrue(EqualsBuilder.reflectionEquals(expected, actual, "id"));
```

Comment:
is it enough to compare only ids?

Reply:
It seems to me that in this case, using EqualsBuilder, I compare the expected ShoppingCartDto with the actual one, excluding the parameter id, and additionally wrapping it in assertTrue
![image](https://github.com/user-attachments/assets/82f47993-61d8-46e6-8b61-15795aee4b6c)